### PR TITLE
GA the health API

### LINKED ISSUE #92209: Promote health API from internal to GA (generally available)
### Description

We've been working on the health API for a while and the breaking changes have slowed to the point where we can promote the API away from the `/_internal` categorization.

### PR DESCRIPTION
**Pull request by andreidan**
This marks the new Health API as generally available.

Fixes: [#92209](https://github.com/elastic/elasticsearch/issues/92209)

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/health/RestGetHealthAction.java [hunk 1]
```diff
@@ -33,7 +33,7 @@ public String getName() {
 
     @Override
     public List<Route> routes() {
-        return List.of(new Route(GET, "/_internal/_health"), new Route(GET, "/_internal/_health/{indicator}"));
+        return List.of(new Route(GET, "/_health"), new Route(GET, "/_health/{indicator}"));
```
Comment (dakrone) [inline server/src/main/java/org/elasticsearch/health/RestGetHealthAction.java:36]: Out of curiosity, should we just *add* `/_health` and `/_health/{indicator}` to the paths so we have a period during which both URLs work? Then at a later time (9.x?) we can remove the `/_internal/health` path?

**Review by dakrone — APPROVED** [review]
LGTM (with all the caveats about synchronizing this with the UI team, unless we decide to make the path parts additive)

File: server/src/main/java/org/elasticsearch/health/RestGetHealthAction.java (hunk 1 above)
Comment (andreidan) [reply server/src/main/java/org/elasticsearch/health/RestGetHealthAction.java:36]: That's a very interesting idea.
I do worry that not ALL our `_internal` APIs will be internal/experimental/not-GA anymore should we choose to do this though. The current invariant of `DO NOT USE` for the `_internal` APIs (irrespective of reading the docs) is quite valuable IMO.

**Comment by crispybacon** [comment]
> 

404 for this link.

**Comment by dakrone** [comment]
@crispybacon the documentation preview links are short-lived and only really for PR review. These docs are live now, and shouldn't need a documentation preview link to access.

**Comment by andreidan** [comment]
Changed this to `>non-issue` as it'll hopefully be avoided from the 8.7 release notes (this PR has been superseded by https://github.com/elastic/elasticsearch/pull/92879 )

**Pull request merged**